sql/src/main/java/io/crate/execution/dsl/projection/SysUpdateProjection.java

```diff
@@ -23,42 +23,98 @@
 package io.crate.execution.dsl.projection;
 
 import io.crate.expression.symbol.InputColumn;
+import io.crate.expression.symbol.SelectSymbol;
 import io.crate.expression.symbol.Symbol;
+import io.crate.expression.symbol.SymbolVisitors;
 import io.crate.expression.symbol.Symbols;
 import io.crate.metadata.Reference;
 import io.crate.metadata.RowGranularity;
-import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.Version;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
+import javax.annotation.Nullable;
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
-public class SysUpdateProjection extends DMLProjection {
+public class SysUpdateProjection extends Projection {
```

Comment:
This could be merged to `UpdateProjection` because the information is redundant and then dispatched using the `ProjectionType` to the designated `Projector`.

Reply:
I'd keep it separate for now, so we've more flexibility with streaming changes in the future.